assets/js/amp-editor-blocks.js

```diff
@@ -0,0 +1,448 @@
+				name = props.name,
+				ampLayout,
+				inspectorControls;
+
+			ampLayout = attributes.ampLayout;
+
+			if ( 'core/shortcode' === name ) {
+				inspectorControls = component.setUpShortcodeInspectorControls( props );
+				if ( '' === inspectorControls ) {
+					// Return original.
+					return [
+						el( BlockEdit, _.extend( {
+							key: 'original'
+						}, props ) )
+					];
+				}
+			} else if ( -1 !== component.data.mediaBlocks.indexOf( name ) || 0 === name.indexOf( 'core-embed/' ) ) {
+				inspectorControls = component.setUpInspectorControls( props );
+			}
+
+			// Return just inspector controls in case of 'nodisplay'.
+			if ( ampLayout && 'nodisplay' === ampLayout ) {
+				return [
+					inspectorControls
+				];
+			}
+
+			return [
+				inspectorControls,
+				el( BlockEdit, _.extend( {
+					key: 'original'
+				}, props ) )
+			];
+		};
+	};
+
+	/**
+	 * Set width and height in case of image block.
+	 *
+	 * @param {Object} props Props.
+	 * @param {string} layout Layout.
+	 */
+	component.setImageBlockLayoutAttributes = function setImageBlockLayoutAttributes( props, layout ) {
+		var attributes = props.attributes;
+		switch ( layout ) {
+			case 'fixed-height':
+				if ( ! attributes.height ) {
+					props.setAttributes( { height: component.data.defaultHeight } );
+				}
+				break;
+
+			case 'fixed':
+				if ( ! attributes.height ) {
+					props.setAttributes( { height: component.data.defaultHeight } );
+				}
+				if ( ! attributes.width ) {
+					props.setAttributes( { width: component.data.defaultWidth } );
+				}
+				break;
+		}
+	};
+
+	/**
+	 * Default setup for inspector controls.
+	 *
+	 * @param {Object} props Props.
+	 * @return {Object|Element|*|{$$typeof, type, key, ref, props, _owner}} Inspector Controls.
+	 */
+	component.setUpInspectorControls = function setUpInspectorControls( props ) {
+		var ampLayout = props.attributes.ampLayout,
+			ampNoLoading = props.attributes.ampNoLoading,
+			isSelected = props.isSelected,
+			name = props.name,
+			el = wp.element.createElement,
+			InspectorControls = wp.editor.InspectorControls,
+			SelectControl = wp.components.SelectControl,
+			ToggleControl = wp.components.ToggleControl,
+			PanelBody = wp.components.PanelBody,
+			label = __( 'AMP Layout' );
+
+		if ( 'core/image' === name ) {
+			label = __( 'AMP Layout (modifies width/height)' );
+		}
+
+		return isSelected && (
+			el( InspectorControls, { key: 'inspector' },
+				el( PanelBody, { title: __( 'AMP Settings' ) },
+					el( SelectControl, {
+						label: label,
+						value: ampLayout,
+						options: component.getLayoutOptions( name ),
+						onChange: function( value ) {
+							props.setAttributes( { ampLayout: value } );
+							if ( 'core/image' === props.name ) {
+								component.setImageBlockLayoutAttributes( props, value );
+							}
+						}
+					} ),
+					el( ToggleControl, {
+						label: __( 'AMP loading indicator disabled' ),
+						checked: ampNoLoading,
+						onChange: function() {
```

Comment:
For some reason, the first time I check the checkbox I get a warning:

> Warning: A component is changing an uncontrolled input of type checkbox to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components

Maybe it's a bug in Gutenberg. It doesn't seem to cause a problem.